masq/src/commands/financials_command.rs

```diff
@@ -58,26 +175,911 @@ impl Command for FinancialsCommand {
+    where
+        F: Fn(&mut dyn Write, &[&str], &[usize]),
+    {
+        let headings_widths = Self::widths_of_dynamical_headings(headings);
+        write_headings(stdout, headings, &headings_widths);
+        short_writeln!(stdout, "\nNo records found",)
+    }
+
+    fn figure_out_max_widths(records: &[&dyn WidthInfo], is_gwei: bool) -> Vec<usize> {
+        let cell_count = records[0].widths(is_gwei).len();
+        let init = vec![0_usize; cell_count];
+        let widths_except_ordinal_num = records.iter().fold(init, |acc, record| {
+            let cells = record.widths(is_gwei);
+            Self::yield_bigger_values_from_vecs(cell_count, acc, cells)
+        });
+        let mut result = vec![records.len().to_string().len()];
+        result.extend(widths_except_ordinal_num);
+        result
+    }
+
+    fn yield_bigger_values_from_vecs(
+        cell_count: usize,
+        first: Vec<usize>,
+        second: Vec<usize>,
+    ) -> Vec<usize> {
+        (0..cell_count).fold(vec![], |acc_inner, idx| {
+            plus(acc_inner, first[idx].max(second[idx]))
+        })
+    }
+
+    fn count_length_with_comma_separators<N: Display>(value: N, is_plain_integer: bool) -> usize {
+        fn add_in_commas_count(integer_length: usize) -> usize {
+            let triple_chars = integer_length / 3;
+            let possible_reminder = integer_length % 3;
+            if possible_reminder == 0 {
+                triple_chars * 3 + triple_chars - 1
+            } else {
+                triple_chars * 3 + possible_reminder + triple_chars
+            }
+        }
+        let string_like = value.to_string();
+        let gross_length = string_like.len();
+        let is_negative = &string_like[0..=0] == "-";
+        let unsigned_num_length = if is_negative {
+            gross_length - 1
+        } else {
+            gross_length
+        };
+        let unsigned_processed = if is_plain_integer {
+            add_in_commas_count(unsigned_num_length)
+        } else {
+            if unsigned_num_length <= 7 {
+                return if !is_negative {
+                    6 //means '< 0.01'
+                } else {
+                    13 //means '-0.01 < x < 0'
+                };
+            } else if unsigned_num_length == 8 {
+                4
+            } else if unsigned_num_length == 9 {
+                4
+            } else {
+                let integer_part_length = unsigned_num_length - 9;
+                add_in_commas_count(integer_part_length) + 3
+            }
+        };
+        unsigned_processed + if !is_negative { 0 } else { 1 }
+    }
+
+    fn triple_blank_line(stdout: &mut dyn Write) {
+        short_writeln!(stdout, "\n\n")
+    }
+}
+
+#[derive(Debug, PartialEq)]
+enum ChangeDone<T> {
+    Changed(T),
+    Unchanged(T),
+}
+
+impl<T> ChangeDone<T> {
+    fn value(self) -> T {
+        match self {
+            Changed(val) => val,
+            Unchanged(val) => val,
+        }
+    }
+}
+
+trait Headings {
```

Comment:
Can this be a struct, used in place of both ReceivableHeadings and PayableHeadings?